configure.py

```diff
@@ -2402,5 +2403,56 @@ def create_building_system(args):
     generate_compdb('compile_commands.json', ninja, args.buildfile, selected_modes)
 
 
+def configure_for_mode(mode, args):
+    # TODO: use Ninja multi-config
+    source_dir = os.path.realpath(os.path.dirname(__file__))
+    build_dir = os.path.join(source_dir, 'build', mode)
+
+    settings = {
+        'CMAKE_BUILD_TYPE': mode,
+        'CMAKE_CXX_COMPILER': args.cxx,
+        'CMAKE_C_COMPILER': args.cc,
+        'CMAKE_C_FLAGS': args.user_cflags,
+        'CMAKE_EXE_LINKER_FLAGS': semicolon_separated(args.user_ldflags),
+        'CMAKE_EXPORT_COMPILE_COMMANDS': 'ON',
+        'CMAKE_EXPORT_NO_PACKAGE_REGISTRY': 'ON',
+    }
+    if args.date_stamp:
+        settings['Scylla_DATE_STAMP'] = args.date_stamp
+
+    cmake_command = ['cmake']
+    cmake_command += [f'-D{var}={value}' for var, value in settings.items()]
+    cmake_command += ['-G', 'Ninja',
+                      '-B', build_dir,
+                      '-S', source_dir]
+    subprocess.check_call(cmake_command, shell=False, cwd=source_dir)
```

Comment:
Shouldn't you create the `build/$mode` dir and use it as work (binary) dir? Or does `CMakeLists.txt` already take care of that?

Reply:
yes, CMake creates the build dir for us.